services/worker/src/worker/job_runners/split/opt_in_out_urls_scan_from_streaming.py

```diff
@@ -160,30 +162,43 @@ async def opt_in_out_scan_urls(
             )  # noqa: E203
         await wait(tasks)
 
-    opt_in_urls_indices = []
-    opt_out_urls_indices = []
+    opt_in_out_urls = []
+    num_opt_in = 0
+    num_opt_out = 0
     for offset, task in zip(offsets, tasks):
-        batch_opt_in_urls_indices, batch_opt_out_urls_indices = task.result()
-        for batch_opt_in_urls_idx in batch_opt_in_urls_indices:
-            opt_in_urls_indices.append(offset + batch_opt_in_urls_idx)
-        for batch_opt_out_urls_idx in batch_opt_out_urls_indices:
-            opt_out_urls_indices.append(offset + batch_opt_out_urls_idx)
-
-    return opt_in_urls_indices, opt_out_urls_indices
+        spawning_response = task.result()
+        spawning_response_urls = spawning_response["urls"]
+        for i in range(len(spawning_response_urls)):
+            if spawning_response_urls[i]["optIn"] or spawning_response_urls[i]["optOut"]:
```

Comment:
each one is called three times, maybe create variables:

```suggestion
            is_opt_in = spawning_response_urls[i]["optIn"]
            is_opt_out = spawning_response_urls[i]["optOut"]
            if is_opt_in or is_opt_out:
```
(or `opt_in` / `opt_out`). Also: should you check the type of is_opt_in and is_opt_out (boolean)?